framework/validators/ExistValidator.php

```diff
@@ -73,6 +73,11 @@ class ExistValidator extends Validator
      */
     public $targetAttributeJunction = 'and';
 
+    /**
+     * @var bool whether this validator can run on salve connections
+     */
+    public $canValidateOnSlaveDb =  true;
```

Comment:
Also I think it should use master connection by default, as that is where the insert will be made it will be safe by default and can be changed if querying slave is okay.